src/component/endorsements/endorsement.jsx

```diff
@@ -0,0 +1,194 @@
+  };
+
+  const suggestions = [
+    "Easy to use",
+    "Requires training",
+    "Informative charts",
+    "Detailed Reports",
+    "Easy reading",
+    "Good for trends"
+  ];
+
+  const handleClick = () => {
+    setOpen(prevState => !prevState);
+  };
+
+  const handleChange = e => {
+    setValue(e.target.value);
+  };
+
+  const Alert = props => {
+    return <MuiAlert elevation={6} variant="filled" {...props} />;
+  };
+
+  const handleValidate = async (field, value) => {
+    let validationError = await validate(field, value);
+    if (validationError) {
+      setError(true);
+      setMessage(validationError);
+    } else {
+      setError(false);
+      setMessage("");
+    }
+  };
+
+  const handleEndorsement = item => {
+    const index = endorsement.indexOf(item);
+    if (index >= 0) {
+      endorsement.splice(index, 1);
+      setEndorsement(preState => [...preState]);
+    }
+    if (value && !error && !message) {
+      const checkValue = endorsement.includes(value);
+      if (!checkValue) {
+        setEndorsement(preState => [...preState, value]);
+        setValue("");
+      } else {
+        handleAlert();
+      }
+    }
+  };
+
+  const keyPress = e => {
+    handleValidate(field, e.target.value);
+    if (!error && !message) {
+      if (e.keyCode === 13) {
+        const value = endorsement.includes(e.target.value);
+        if (!value) {
+          setEndorsement(preState => [...preState, e.target.value]);
+          setValue("");
+        } else {
+          handleAlert();
+        }
+      }
+    }
+  };
+
+  const addSuggestions = item => {
+    setValue(item);
+  };
+
+  const renderChips = () => {
+    if (endorsement.length > 0) {
+      const newEndorsement = endorsement.slice(0, 2);
+      let count;
+      if (endorsement.length > 2) {
+        count = parseInt(endorsement.length) - 2;
+      }
+      if (count && !open) {
+        return (
+          <>
+            {newEndorsement.map((item, index) => (
+              <Chip
+                size="small"
+                icon={<AddIcon fontSize="inherit" />}
+                label={item}
+                id={`${index}small`}
+                key={`${index}small`}
+                style={{ margin: 5 }}
+              />
+            ))}
+            {count && (
+              <Chip
+                size="small"
+                icon={<AddIcon fontSize="inherit" />}
+                label={`${count}more`}
+                onClick={handleClick}
+              />
+            )}
+          </>
+        );
+      }
+      return (
+        <>
+          {endorsement.map((item, index) => (
+            <Chip
+              size="small"
+              icon={<AddIcon fontSize="inherit" />}
+              label={item}
+              id={`${index}small`}
+              key={`${index}small`}
+              onClick={value => handleEndorsement(item)}
+              style={{ margin: 5 }}
+            />
+          ))}
+        </>
```

Comment:
a simple function rendering a chip component controlled by passing dynamic props can be used instead of using chip component every time